apis/core.oam.dev/common/types.go

```diff
@@ -336,7 +336,8 @@ type WorkflowStatus struct {
 	Steps          []workflowv1alpha1.WorkflowStepStatus `json:"steps,omitempty"`
 
 	StartTime metav1.Time `json:"startTime,omitempty"`
-	EndTime   metav1.Time `json:"endTime,omitempty"`
+	// +nullable
+	EndTime metav1.Time `json:"endTime,omitempty"`
```

Comment:
is Null reasonable?

Reply:
Is the because the Application exists before controller upgradation so that the `endTime` is null?

Reply:
My application is new created, If the workflow is not completed  eg:running, no endtime at this time.
caused err:
E0106 08:12:02.807341       1 controller.go:317] controller/application "msg"="Reconciler error" "error"="Application.core.oam.dev \"test\" is invalid: status.workflow.endTime: Invalid value: \"null\": status.workflow.endTime in body must be of type string: \"null\"" "name"="test" "namespace"="test" "reconciler group"="core.oam.dev" "reconciler kind"="Application"

Reply:
I think the endtime can be null if the workflow is not completed

Reply:
What's your Kubernetes version? Maybe it's related to the Kubernetes version.

Reply:
@FogDong v1.19